src/nifreeze/viz/motion_viz.py

```diff
@@ -0,0 +1,267 @@
+#
+
+from typing import Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.axes import Axes
+from scipy.ndimage import gaussian_filter
+
+
+def _get_axis(orientation: str):
+    """Get axis for the given orientation.
+
+    Parameters
+    ----------
+    orientation : :obj:`str`
+        Orientation. Cn be one of ``axial``, ``coronal``, or ``sagittal``.
+
+    Returns
+    -------
+    :obj:`int`
+        Axis corresponding to the given orientation.
+    """
+
+    if orientation == "axial":
+        return 2
+    elif orientation == "coronal":
+        return 1
+    elif orientation == "sagittal":
+        return 0
+    else:
+        raise ValueError(
+            f"Orientation must be one of 'axial', 'coronal', 'sagittal'. Provided: {orientation}"
+        )
+
+
+def _extract_slice(img_data: np.ndarray, orientation: str, slice_idx: int):
+    """Extract slice data from the given volume at the given orientation slice index.
+
+    Parameters
+    ----------
+    img_data : :obj:`~numpy.ndarray`
+        Image data to be sliced.
+    orientation : :obj:`str`
+        Orientation.
+    slice_idx : :obj:`int`
+        Slice index.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Image slice.
+    """
+
+    axis = _get_axis(orientation)
+
+    axis_sizw = img_data.shape[axis]
+
+    if not (0 <= slice_idx < axis_sizw):
+        raise IndexError(
+            f"Slice index {slice_idx} out of bounds for axis {orientation} with size {axis_sizw}"
+        )
+
+    if axis == 2:
+        return img_data[:, :, slice_idx]
+    elif axis == 1:
+        return np.rot90(img_data[:, slice_idx, :])
+    else:
+        return np.rot90(img_data[slice_idx, :, :])
+
+
+def plot_framewise_displacement(
+    fd1: np.ndarray,
+    fd2: np.ndarray,
+    label1: str,
+    label2: str,
+    ax: Union[Axes, None] = None,
+) -> Axes:
+    """Plot frame-wise displacement data.
+
+    Plots the frame-wise displacement data corresponding to two different
+    realizations.
+
+    Parameters
+    ----------
+    fd1 : :obj:`~numpy.ndarray`
+        Frame-wise displacement values corresponding to first realization.
+    fd2 : :obj:`~numpy.ndarray`
+        Frame-wise displacement values corresponding to second realization.
+    label1 : :obj:`str`
+        Label for the first realization legend.
+    label2 : :obj:`str`
+        Label for the second realization legend.
+    ax : :obj:`Axes`, optional
+        Figure axes.
+
+    Returns
+    -------
+    ax : :obj:`Axes`
+        Figure plot axis.
+    """
+
+    fd1_dims = fd1.ndim
+    fd2_dims = fd2.ndim
+
+    n_frames_fd1 = fd1.shape
+    n_frames_fd2 = fd2.shape
+
+    req_dims = 1
+    if fd1_dims != req_dims or fd2_dims != req_dims:
+        raise ValueError(f"fd1 and fd2 must be 1D: {n_frames_fd1}; {n_frames_fd2}")
+
+    if n_frames_fd1 != n_frames_fd2:
+        raise ValueError(f"fd1 and fd2 must have same shape: {n_frames_fd1}; {n_frames_fd2}")
```

Comment:
If they come in a single numpy 2D array, this is not necessary anymore.

Reply:
Rmoved in e8fde0c. The input is now a pandas datraframe.